Scholars Portal Journals.js

```diff
@@ -0,0 +1,172 @@
+{
+	"translatorID": "5ac0fd37-5578-4f82-8340-0e135b6336ee",
+	"label": "Scholars Portal Journals",
+	"creator": "Bartek Kawula",
+	"target": "https?://journals[1-2]\\.scholarsportal\\.info/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2016-02-19 04:18:37"
+}
+
+function detectWeb(doc, url) { 
+	// see if saved list is toggled open
+	if (doc.getElementsByClassName('inner-wrap-open') [0]) {
+		return 'multiple'
+	} else {
+		if(url.indexOf("/search?q") != -1 || url.indexOf("/browse/") != -1){
+			return 'multiple'
+		}
+		else {
+			return 'journalArticle';
+		}
+	}
+}
+
+function doWeb(doc, url) {
+	var type = detectWeb(doc, url);
+	if(type == "multiple") {
+		var list = getItems(doc, url);
+		Zotero.selectItems(list, function(selectedItems) {
+			if(!selectedItems) return true;
+			var uri = [];
+			for(var i in selectedItems) {
+				uri.push(list[i].uri)
+			}
+			for(var i=0; i<uri.length; i++) {
+				var url = uri[i].replace(/(\/details)/g,"");
+				scrape(doc, url)
+			}
```

Comment:
merge the two loops, but you should collect all the URLs together first, then call `ZU.doGet` on all and `scrape` the resulting text. This performs sequential HTTP requests, rather than spamming the servers.